src/components/clusterprovider/common/cacheinfo.ts

```diff
@@ -0,0 +1,53 @@
+import { getMinikubeShowInfoState } from "../../config/config";
+
+export enum ShowInformationOptions {
+    Install = "Install",
+    InstallFortNightly = "Remind me in 2 weeks",
+    InstallHalfYearly = "Remind me in 6 months",
+}
+
+/*
+    This work is done specifically from the feedback from community for minikube or helm.
+
+    The idea is that for those 2 we will have custom dialog boxes with options to be reminder dialog box,
+    in every x-days.
+
+    Now for example: Now when "Minikube upgrade available to x.xx, currently on x.xy" informationMessage
+    we display user with 3 buttons to choose from:
+
+    Option 1: Install (which is normal behaviour)
+    Option 2: Install fortnightly - which we will cache as expiration date.
+    Option 3: Install Half Yearly - which we will cache as expiration date.
+
+    At the extension.ts level we check if there is a cached value for the minikubereminder, if so and its
+    currently valid, then no message will be displayed otherwise show informaionmessage will popup for the end user.
+*/
+
+export async function isMinikubeInfoDisplay() {
+    const showInfoExpiration = await getMinikubeShowInfoState();
+    return isCacheExpired(showInfoExpiration);
+}
+
+export function getCacheExpirationDate(expiration: string) {
+    let expirationDate = new Date();
+    switch (expiration) {
+        case ShowInformationOptions.InstallFortNightly:
+            expirationDate.setTime(expirationDate.getTime() + 15 * 24 * 3600000);
+            return expirationDate;
+        case ShowInformationOptions.InstallHalfYearly:
+            expirationDate = new Date();
+            expirationDate.setTime(expirationDate.getTime() + 180 * 24 * 3600000);
+            return expirationDate;
+        default:
+            console.log("No cache expiration exists!");
+            return;
+    }
+}
+
+function isCacheExpired(expiration: string | undefined): boolean {
+    if (expiration === undefined || !expiration) {
+        return true;
+    }
+
+    return new Date(expiration) <= new Date();
```

Comment:
Does it behave correctly when `new Date(expiration)` returns an invalid date? Expiration could have an unknown value as this could be errouneously changed by the user

Reply:
❤️❤️☕️ Thank you for this @lstocchi , done, update and added tests for date validation to avoid any manual input mistake and serve user with message and dialog box.

All tests passing: (Please take a look)

<img width="276" alt="Screenshot 2022-11-03 at 5 45 42 AM" src="https://user-images.githubusercontent.com/6233295/199550965-f40478fd-f162-4fe4-af94-0c85258b2bc3.png">